homeassistant/components/tmb/sensor.py

```diff
@@ -0,0 +1,113 @@
+"""Support for TMB (Transports Metropolitans de Barcelona) Barcelona public transport."""
+from datetime import timedelta
+
+from tmb import IBus
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+ATTRIBUTION = "Data provided by Transport Metropolitans de Barcelona"
+
+ICON = "mdi:bus-clock"
+
+CONF_APP_ID = "app_id"
+CONF_APP_KEY = "app_key"
+CONF_LINE = "line"
+CONF_BUS_STOP = "stop"
+CONF_BUS_STOPS = "stops"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+LINE_STOP_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_BUS_STOP): cv.string,
+        vol.Required(CONF_LINE): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+    }
+)
+
+ROUTES_SCHEMA = vol.All(cv.ensure_list, [LINE_STOP_SCHEMA])
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_APP_ID): cv.string,
+        vol.Required(CONF_APP_KEY): cv.string,
+        vol.Optional(CONF_BUS_STOPS): ROUTES_SCHEMA,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the sensors."""
+    ibus_client = IBus(config[CONF_APP_ID], config[CONF_APP_KEY])
```

Comment:
I assume that there will be a traceback if the credentials are wrong. You should try to catch that exception and make the platform setup fails otherwise will the users end up with a non-working integration in their setup.

Reply:
good one, let me check

Reply:
Any news?

Reply:
There is not an endpoint for checking that the API keys are valid or not, so I configured an exception that will raised if the forecast calls to the API fail.